### DISCUSSION TITLE
How to upgrade an EdgeDB Docker container image version when using an external Postgres database?

### DISCUSSION THREAD

**Discussion by elliotwaite** [geldata/gel#6645] [Q&A] [answered]
If I'm running EdgeDB in a Docker container that connects to an external Postgres database, do I have to do anything special when upgrading the EdgeDB Docker image version?

For example, if I'm running a container that is using the `edgedb/edgedb:4.2` image, can I just stop that container and bring up a new container running the `edgedb/edgedb:4.3` image and have it connect to the same external Postgres database that the previous container was connected to? And if so, how is this handled by the EdgeDB server? Does it do something like store the current EdgeDB server version in the database, and then when the newer version of the EdgeDB server starts up, it detects that the version has changed and it makes any necessary changes to the database before proceeding?

**Comment by elprans** [comment, marked as answer]
For minor version upgrades (e.g 4.2 to 4.3) you don't need to do anything special, the EdgeDB server will perform necessary upgrades (if any) on first launch.  Major version upgrades currently require that you use a fresh Postgres database and do an `edgedb dump/restore` from an earlier EdgeDB version.

**Reply by elliotwaite** [reply]
Okay, got it. Thanks for the quick reply. And just to clarify, can it handle any version jump as long as the major version is the same (e.g. 4.0 to 4.3)?

Also, can it automatically handle minor version downgrades, for example, if I'm experiencing an issue with 4.3 and want to downgrade back to 4.2? Or would I have to do an `edgedb dump/restore` in that case?

**Reply by elprans** [reply]
> And just to clarify, can it handle any version jump as long as the major version is the same (e.g. 4.0 to 4.3)?

Yes

> Also, can it automatically handle minor version downgrades, for example, if I'm experiencing an issue with 4.3 and want to downgrade back to 4.2?

Generally, yes, as the minor version patches are generally backwards compatible, but we don't explicitly test things in that direction, so it's always best to test the upgrade on a non-prod deployment.  It's also a good idea to make an `edgedb dump` (or a Postgres-level snapshot) before you do any upgrade.

**Reply by elliotwaite** [reply]
Okay, makes sense. Thanks for the info!
